docs/support/ghc-version-support.md

```diff
@@ -76,12 +80,43 @@ For example, if 9.2 is the latest major version fully supported by HLS and used
 
 HLS will support all non-legacy major versions of GHC.
 
-### Minor versions
+#### Minor versions
 
 For the latest supported major GHC version we will support at least 2 minor versions.
 
 For the rest of the supported major GHC versions, we will support at least the latest minor version in Stackage LTS (so 1 minor version).
 
+### Extended policy
+
+This is the part of the policy that needs evaluation by a human and possibly followed
+by a discussion.
+
+#### Ecosystem factors
+
+To establish and apply the policy we also take into account:
+
+- Completeness: support includes all plugins and features
+- The most recent [stackage](https://www.stackage.org/) LTS snapshot
+- The GHC versions used in the most popular [linux distributions](https://repology.org/project/ghc/versions)
+- The reliability of different ghc versions on the major operating systems (Linux, Windows, MacOS)
+- The [Haskell Survey results](https://taylor.fausak.me/2022/11/18/haskell-survey-results/#s2q4)
+
+### Supporting a GHC version beyond its shelf life
+
+In cases where the base policy demands a deprecation, but ecosystem factors
+suggest that it's still widely used (e.g. last [Haskell Survey results](https://taylor.fausak.me/2022/11/18/haskell-survey-results/#s2q4)), the deprecation should be suspended for the next release and the situation be re-evaluated for the release after that.
```

Comment:
```suggestion
Sometimes the policy suggests that we should deprecate a version of GHC, but doing so would be particularly harmful for our users because it remains very widely used. In cases like this we can decide to keep supporting that GHC version, but this decision should be revisited each time we do a release.

To determine whether we should keep supporting a GHC version we can use 
- The GHC versions used in the most popular [linux distributions](https://repology.org/project/ghc/versions)
- The [Haskell Survey results](https://taylor.fausak.me/2022/11/18/haskell-survey-results/#s2q4)
- Anecdotal evidence about he reliability of different GHC versions on the major operating systems (Linux, Windows, MacOS), e.g. noting the presence of serious bugs
```

Reply:
I think this is mixing up things. Ecosystem factors should be its own section and be referenced from this section explaining the **process** (not criteria) of supporting an older GHC version.

Reply:
I agree and think this separation helps the reader understanding what is going on.